platforms/common/include/px4_platform_common/spi.h

```diff
@@ -64,8 +64,8 @@ struct px4_spi_bus_devices_t {
 
 struct px4_spi_bus_t {
 	px4_spi_bus_device_t devices[SPI_BUS_MAX_DEVICES];
-	int bus{-1}; ///< physical bus number (1, ...) (-1 means this is unused)
 	uint32_t power_enable_gpio{0}; ///< GPIO (if non-zero) to control the power of the attached devices on this bus (0 means power is off)
+	int8_t bus{-1}; ///< physical bus number (1, ...) (-1 means this is unused)
```

Comment:
@dagar let me know if this looks sane.

I could reduce flash by about 3000 bytes by using uint8_t for `power_enable_gpio` as well as the `spi_drdy_gpio_t`, and `cs_gpio` but stuff stopped working, so that probably went too far.

I also wonder whether we should set `SPI_BUS_MAX_DEVICES` based on the actual configuration somehow to reduce the footprint of this.

Also, some of the SPI configurations are completely identical but given the data structure and `constexpr` generation we have to duplicate it. Any idea how to work around that?